flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSchemaConverter.java

```diff
@@ -362,7 +369,11 @@ public static Schema convertToSchema(LogicalType logicalType, String rowName) {
 					.record(rowName)
 					.fields();
 				for (int i = 0; i < rowType.getFieldCount(); i++) {
-					String fieldName = rowName + "_" + fieldNames.get(i);
+					String fieldName = fieldNames.get(i);
+					if (rowName.equals(fieldName)) {
+						// Can not build schema when the record and field have the same name
+						fieldName = rowName + "_" + fieldName;
+					}
```

Comment:
I think there is still existed a bug. see: https://issues.apache.org/jira/browse/FLINK-19779?focusedCommentId=17219588&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17219588

Reply:
Yes, the avro schema builder does not allow same name field names, even if they are in different scope (different layer).

Reply:
I think the correct solution will be:

```
				RowType rowType = (RowType) logicalType;
				List<String> fieldNames = rowType.getFieldNames();
				// we have to make sure the record name is different in a Schema
				SchemaBuilder.FieldAssembler<Schema> builder =
						getNullableBuilder(logicalType)
								.record(rowName)
								.fields();
				for (int i = 0; i < rowType.getFieldCount(); i++) {
					String fieldName = fieldNames.get(i);
					builder = builder
						.name(fieldName)
						.type(convertToSchema(rowType.getTypeAt(i), rowName + "_" + fieldName))
						.noDefault();
				}
				return builder.endRecord();
```